src/openms/include/OpenMS/DATASTRUCTURES/FASTAContainer.h

```diff
@@ -350,6 +350,15 @@ class DecoyHelper
     bool is_prefix; ///< on success, was it a prefix or suffix
   };
 
+  struct DecoyData
+  {
+    std::unordered_map<std::string, std::pair<Size, Size>> decoy_count; ///< map decoys to counts of occurrences as prefix/suffix
+    std::unordered_map<std::string, std::string> decoy_case_sensitive; ///< map case insensitive strings back to original case (as used in fasta)
+    Size all_prefix_occur; ///< number of proteins with found decoy_prefix
+    Size all_suffix_occur; ///< number of proteins with found decoy_suffix
+    Size all_proteins_count; ///< number of all checked proteins
+  };
+
   // decoy strings
   inline static const std::vector<std::string> affixes = { "decoy", "dec", "reverse", "rev", "reversed", "__id_decoy", "xxx", "shuffled", "shuffle", "pseudo", "random" };
```

Comment:
Does this need to be a Vector? You only have to read the strings and do not mutate them. So maybe a constant string array would be more performance oriented. 
```suggestion
  inline static const std::string[11] affixes = { "decoy", "dec", "reverse", "rev", "reversed", "__id_decoy", "xxx", "shuffled", "shuffle", "pseudo", "random" };
```

Reply:
> Does this need to be a Vector? You only have to read the strings and do not mutate them. So maybe a constant string array would be more performance oriented.

Isnt the object sitting on the stack anyways?

Reply:
Has to be a vector because the following function (ListUtils::concatenate) working with affixes needs a vector as input.